grammar/ja/src/rules.rs

```diff
@@ -173,37 +174,45 @@ pub fn rules_percentage(b: &mut RuleSetBuilder<Dimension>) -> RustlingResult<()>
 }
 
 pub fn rules_finance(b: &mut RuleSetBuilder<Dimension>) -> RustlingResult<()> {
-    b.rule_2("intersect (X cents)",
+    b.rule_2("intersect <money> (X cents)",
              amount_of_money_check!(),
              amount_of_money_check!(|money: &AmountOfMoneyValue| money.unit == Some("cent")),
              |a, b| helpers::compose_money(a.value(), b.value()));
-    b.rule_2("intersect",
+    b.rule_2("intersect <money>",
              amount_of_money_check!(|money: &AmountOfMoneyValue| money.unit != Some("cent")),
              number_check!(),
              |a, b| helpers::compose_money_number(&a.value(), &b.value()));
     b.rule_1_terminal("USD",
+        b.reg(r#"アメリカドル"#)?,
+        |_| Ok(MoneyUnitValue { unit: Some("USD") })
+    );
+    b.rule_1_terminal("$",
         b.reg(r#"ドル|\$|＄"#)?,
         |_| Ok(MoneyUnitValue { unit: Some("$") })
     );
     b.rule_1_terminal("EUR",
         b.reg(r#"ユーロ|€"#)?,
         |_| Ok(MoneyUnitValue { unit: Some("€") })
     );
-    b.rule_1_terminal("GBP",
-        b.reg(r#"ポンド|£"#)?,
+    b.rule_1_terminal("£",
+        b.reg(r#"£"#)?,
         |_| Ok(MoneyUnitValue { unit: Some("£") })
     );
+    b.rule_1_terminal("GBP",
+        b.reg(r#"ポンド"#)?,
+        |_| Ok(MoneyUnitValue { unit: Some("GBP") })
+    );
     b.rule_1_terminal("JPY",
         b.reg(r#"円|¥"#)?,
```

Comment:
A rule should be created for ¥ as it is the same symbol for JPY and CNY (see below)